src/library/baseplaylistfeature.cpp

```diff
@@ -478,6 +484,20 @@ void BasePlaylistFeature::slotExportPlaylist() {
     }
 }
 
+void BasePlaylistFeature::slotExportTrackFiles() {
+    int rows = m_pPlaylistTableModel->rowCount();
+    QList<TrackPointer> tracks;
+    for (int i = 0; i < rows; ++i) {
+        QModelIndex index = m_pPlaylistTableModel->index(i, 0);
+        tracks.push_back(m_pPlaylistTableModel->getTrack(index));
+    }
+
+    DlgTrackExport track_export_dlg(nullptr, m_pConfig, tracks);
```

Comment:
This is a place where a new controller class can do its work. 
It simply looks weired to hide all business logic in a Dlg Class here.

How about a TrackExportCtrl that holds the DlgTrackExport and the TrackExport thread.
It has not much, basically only the content of selectDestinationDirectory()
This would also solve the dialog in dialog issue issue that DlgTrackExport opens the file dialog before the original TrackDialog is shown. It would also allow more testing since you can mock out the dialogs.